src/qwed_new/core/diagnostics.py

```
@@ -727,6 +728,34 @@ def enforce_trust_decision(
 
+    # Issue #273 — TOCTOU: detach the caller's mutable developer_fields before
+    # any validation reads the result. DiagnosticResult is frozen=True, but
+    # developer_fields is a deeply mutable Dict[str, Any]; returning the
+    # caller's original reference would let out-of-band mutation after
+    # enforcement diverge the returned state from the validated state. The
+    # snapshot below is read during validation AND returned, so a concurrent
+    # mutation of the caller's object can neither skew the decision nor leak
+    # into the returned result.
+    try:
+        result = replace(result, developer_fields=deepcopy(result.developer_fields))
+    except Exception as exc:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)